terraform/graph_config_node_variable.go

```diff
@@ -130,17 +130,22 @@ func (n *GraphNodeConfigVariable) hasDestroyEdgeInPath(opts *NoopOpts, vertex da
 	if vertex == nil {
 		vertex = opts.Vertex
 	}
+
+	var root graphNodeRoot
```

Comment:
Yeah, checking that c is a root node is a little cleaner. Unless we had a single global rootNode value to compare to, I agree the struct comparison felt weird. I'll update